### PR TITLE
ATT-38: AttachmentBytesResource_* to take 'view' as parameter.

### PR DESCRIPTION
**Pull request by reagan-meant**
https://issues.openmrs.org/browse/ATT-38

Enabled the URI to take in a new param called `view` to allow the `AttachmentBytesResource` to get files depending on the view or default if null.

### CODE REVIEW

File: omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10.java [hunk 1]
```diff
@@ -37,7 +38,7 @@
 	protected final Log log = LogFactory.getLog(getClass());
 	
 	@RequestMapping(value = AttachmentsConstants.ATTACHMENT_BYTES_URI, method = RequestMethod.GET)
-	public void getFile(@PathVariable("uuid") String uuid, HttpServletResponse response) throws ResponseException {
+	public void getFile(@PathVariable("uuid") String uuid,@RequestParam(required = false, value = "view") String view, HttpServletResponse response) throws ResponseException {
```
Comment (mks-d) [inline omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10.java]: A detail but this:
```java
String uuid,@RequestParam(required =
```
so the missing blank space between "`uuid`" and "`@RequestParam`" shows that the file is misformatted. In other words you did not run `mvn clean package` before pushing.

Always run `mvn clean package` before pushing code to GitHub.

**Review by mks-d — CHANGES_REQUESTED** [review]
Thanks @reagan-meant! 

This was the easy part though, now the controller test(s) that show(s) that all goes as expected.

File: omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10.java (hunk 1 above)
Comment (reagan-meant) [reply omod-1.10/src/main/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10.java]: Thanks for this.

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	public void tearDown() throws IOException {
+		testHelper.tearDown();
+	}
+	
+	@Test
+	public void getAttachmentBytes_shouldGetOriginalFileWhenViewPassedEqualToOriginal() throws Exception {
+		
+		// Setup
+		String fileCaption = "Test file caption";
+		String uuid;
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: Instantiate `uuid` where you first use it. (Do the same for all test cases.)

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		String fileName = "testFile." + fileExtension;
+		
+		// Upload Test File
+		Patient patient = Context.getPatientService().getPatient(2);
+		MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, mimeType, randomData);
+		
+		uploadRequest.addFile(file);
+		uploadRequest.addParameter("patient", patient.getUuid());
+		uploadRequest.addParameter("fileCaption", fileCaption);
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: You never reuse `fileCaption`, so don't make it a variable. Just do:
```java
uploadRequest.addParameter("fileCaption", "Test file caption");
```

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		
+		SimpleObject uploadResponse = deserialize(handle(uploadRequest));
+		uuid = uploadResponse.get("uuid");
+        
+		HttpServletRequest downloadRequest = newGetRequest(getURI() + "/" + uuid + "/bytes",new Parameter("view", "complexdata.view.original"));
```
Comment (mks-d) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: Put this in the replay section.

**Review by mks-d — CHANGES_REQUESTED** [review]
Thanks @reagan-meant.

We will need to do different test(s) ;-)

The issue here is that you never validate that you get the complex data (so the content bytes in your case) that corresponds to a different view. Whatever you do it's always the same bytes that come back.

I'll try to provide some guidance on the ticket if/when I find the time.

File: omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+			
+			// Verify
+			Assert.assertArrayEquals(randomData, bytesContent);
+			Assert.assertEquals(downloadResponse.getContentType(), mimeType);
+			Assert.assertEquals(downloadResponse.getHeader("File-Name"), fileName);
+			Assert.assertEquals(downloadResponse.getHeader("File-Ext"), fileExtension);
+			
+		}
+		@Test
+		public void getAttachmentBytes_shouldGetThumbNailFileWhenViewIsPassedEqualToThumbnail() throws Exception {
```
Comment (reagan-meant) [inline omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: @mks-d I was under the impression that this tests to see the returned bytesContent when we try with the thumbnail view which is zero as compared to the originalview

Comment (mks-d) [reply omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: Correct @reagan-meant, I missed that. I will review this again.

Comment (reagan-meant) [reply omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentBytesResource1_10Test.java]: @mks-d I will be waiting for ur feedback

**Review by mks-d — APPROVED** [review]

**Pull request merged**